src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift

```diff
@@ -133,6 +136,10 @@ class HealthCertificateService {
 		Log.info("[HealthCertificateService] Registering health certificate from payload: \(private: base45)", log: .api)
 
 		do {
+			if case .failure = signatureVerifying.verify(certificate: base45, with: [], and: Date()) {
+				return .failure(.invalidSignature)
+			}
```

Comment:
I think the signature check is too early here. According to the tech spec we should 1. Check and remove prefix, 2. Decode, 3. Verify signature, 4. Check JSON schema, 5. Check content, ...

I think 1, 2 and 4 are actually all contained in the `try HealthCertificate(base45: base45)`, so maybe check the signature right after that?